src/aspire/source/__init__.py

```diff
@@ -288,28 +288,27 @@ def eval_filters(self, im_orig, start=0, num=np.inf, indices=None):
         if indices is None:
             indices = np.arange(start, min(start + num, self.n))
 
-        unique_filters = set(self.filters)
-        for f in unique_filters:
-            idx_k = np.where(self.filters[indices] == f)[0]
+        for f_ind in range(len(self.unique_filters)):
+            idx_k = np.where(self.filter_indices[indices] == f_ind)[0]
             if len(idx_k) > 0:
-                im[idx_k] = Image(im[idx_k]).filter(f).asnumpy()
+                im[idx_k] = Image(im[idx_k]).filter(self.unique_filters[f_ind]).asnumpy()
 
         return im
 
     def eval_filter_grid(self, L, power=1):
         grid2d = grid_2d(L)
         omega = np.pi * np.vstack((grid2d['x'].flatten(), grid2d['y'].flatten()))
 
-        h = np.empty((omega.shape[-1], len(self.filters)))
-        for f in set(self.filters):
-            idx_k = np.where(self.filters == f)[0]
+        h = np.empty((omega.shape[-1], len(self.filter_indices)))
+        for f_ind in range(len(self.unique_filters)):
```

Comment:
In cases where you're iterating through `range(len(self.unique_filters))`, you should instead do an `enumerate(self.unique_filters)`. That way you have both the filter and its index at once.

Reply:
Changed to use enumerate.